overlord/snapstate/snapstate.go

```diff
@@ -314,10 +318,43 @@ func Remove(s *state.State, name string) (*state.TaskSet, error) {
 	return full, nil
 }
 
-// Rollback returns a set of tasks for rolling back a snap.
+// Revert returns a set of tasks for reverting to the pervious version of the snap.
 // Note that the state must be locked by the caller.
-func Rollback(s *state.State, snap, ver string) (*state.TaskSet, error) {
-	return nil, fmt.Errorf("rollback not implemented")
+func Revert(s *state.State, name string) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+
+	pi := snapst.PreviousSideInfo()
+	if pi == nil {
+		return nil, fmt.Errorf("no revision to revert to")
+	}
+	return revertToRevision(s, name, pi.Revision)
+}
+
+func revertToRevision(s *state.State, name string, rev snap.Revision) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+
+	if !snapst.Active {
+		return nil, fmt.Errorf("cannot revert inactive snaps")
+	}
```

Comment:
not an issue atm but we should check if we are not trying to revert to snapst.Current and either do nothing? or error?